File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.LinkedList;
import java.util.Queue;
    
class Solution {
    
    static void bfs(int[][] computers, int start, boolean[] visited) {
        Queue<Integer> queue = new LinkedList<>();
        queue.add(start);
        visited[start] = true;

		int n = computers.length;

        while (!queue.isEmpty()) {
            int currNode = queue.poll();

            for (int neighbor = 0; neighbor < n; neighbor++) {
                if (computers[currNode][neighbor] == 1 && !visited[neighbor]) {
                    queue.add(neighbor);
                    visited[neighbor] = true;
                }
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        boolean[] visited = new boolean[n];
        
        int networkCount = 0;
        
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                bfs(computers, i, visited);
                networkCount++;
            }
        }
        
        return networkCount;
   }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    visited = [False] * n
    
    def bfs(start):
        queue = deque([start])
        visited[start] = True
        
        while queue:
            curr_node = queue.popleft()
            
            for neighbor in range(n):
                if computers[curr_node][neighbor] == 1 and not visited[neighbor]:
                    queue.append(neighbor)
                    visited[neighbor] = True
            
    
    network_count = 0
    for i in range(n):
        if not visited[i]:
            bfs(i)
            network_count += 1
    
    return network_count